lib/SIL/Utils/DynamicCasts.cpp

```diff
@@ -115,8 +115,14 @@ classifyDynamicCastToProtocol(CanType source, CanType target,
 
   // If checkConformance() returns a valid conformance, then all conditional
   // requirements were satisfied.
-  if (checkConformance(source, TargetProtocol))
+  if (auto conformance = checkConformance(source, TargetProtocol)) {
+    if (conformance.isConcrete()) {
+      if (!matchesActorIsolation(conformance.getConcrete(), function))
+        return DynamicCastFeasibility::MaySucceed;
```

Comment:
You also need to check whether any conditional requirements have been satisfied by isolated conformances here. `ProtocolConformanceRef::forEachIsolatedConformance` knows how to enumerate all of the isolated conformances, including those satisfying conditional requirements. You could build on that determine the isolation of a given `ProtocolConformanceRef` and use that instead of the isConcrete check.